### ISSUE TITLE
MinGW build fails in Appveyor

### ISSUE THREAD

**Issue by osechet** [conan-io/conan-package-tools#36]
This [MinGW build](https://ci.appveyor.com/project/osechet/conan-qt-base/build/job/0n6nw48w1rkjgwfb) failed in Appveyor but I don't know why. Everything works fine until the package creation. Then it complains about virtual environment. I have never seen this problem until now. I guess it's related to to the fact conan-package-tools uses mingw_installer (in a virtual environment) but I don't know what to do to fix it.
Can you help me?

**Comment by lasote** [comment]
Hi, When I generate the contents for the activate.bat I look if the _CONAN_VENV env variable is defined. If it's defined I raise the error that you are seeing: "Deactivate the current virtual environment..." It's not ideal but we wanted to prevent the environment getting dirty with multiple executions of the virtual environment.

The problem is, I don't know why is keeping the variable _CONAN_VENV. It is set when you run the activate.bat, but I assume that a "self.run("activate.bat  && " invocation should not alter the system environment (just the current process one), but it seems that in some point is keeping it. Maybe keeping the python process one? We need to test it and see what happen. 

If you want to help debugging maybe you can try crazy things like:

```
self.run("activate && %s %s && SET _CONAN_VENV=" % (os.sep.join([".", "bin", "helloworld"]), "conan"))
```

**Comment by lasote** [comment]
Hi, I think the problem is conan_package_tools is activating a virtualenv for compile with the custom MinGW and then it complains for your test method. I have to think the better fix to this situation, but now I understand it.

**Comment by lasote** [comment]
Hi. I've opened a PR in conan repository. We are going to remove those checks, I think they are not necessary anymore before some improvements in last releases. Will be fixed in release 0.16
